composer/loggers/slack_logger.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log metrics to slack, using Slack postMessage api."""
+import os
+import time
+from typing import Any, Callable, Dict, List, Optional, Set
+
+from composer.loggers.logger_destination import LoggerDestination
+from composer.utils import MissingConditionalImportError
+
+
+class SlackLogger(LoggerDestination):
+    """Log metrics to slack, using Slack's postMessage api - https://api.slack.com/methods/chat.postMessage.
+
+    First export 2 environment variable to use this logger.
+    1. SLACK_LOGGING_API_KEY: To get app credentials, follow tutorial here - https://api.slack.com/tutorials/tracks/posting-messages-with-curl?app_id_from_manifest=A053W1QCEF2.
+    2. SLACK_LOGGING_CHANNEL_ID: Channel id to send the message (Open slack channel in web browser to look this up).
+
+    Next write script to output metrics / hparams / traces to slack channel. See example below.
+
+    .. code-block:: python
+
+        trainer = Trainer(
+            model=mnist_model(num_classes=10),
+            train_dataloader=train_dataloader,
+            max_duration='2ep',
+            algorithms=[
+                LabelSmoothing(smoothing=0.1),
+                CutMix(alpha=1.0),
+                ChannelsLast(),
+            ],
+            loggers=[
+                SlackLogger(
+                    formatter_func=(lambda data, **kwargs: [{
+                        'type': 'section',
+                        'text': {
+                            'type': 'mrkdwn',
+                            'text': f'*{k}:* {v}'
+                        }
+                    } for k, v in data.items()]),
+                    include_keys={'loss/train/total'},
+                    interval_in_seconds=1
+                ),
+            ],
+        )
+
+        trainer.fit()
+
+    Args:
+        formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack.
+        include_keys (Set[str]): A set of metric/logs/traces keys to include in the message. If None, all keys are included.
+        interval_in_seconds (int): Interval in seconds to send logs to slack. If None, message is sent after every log call.
+        max_logs_per_message (int)(default:50): Maximum number of logs to send in a single message. Note that no more than 50 items are allowed to send in a single message.
+        If more than 50 items are stored in buffer, the message flushed without waiting the full time interval.
+    """
+
+    def __init__(
+        self,
+        formatter_func: Optional[Callable[..., List[Dict[str, Any]]]] = None,
+        include_keys: Optional[Set[str]] = None,
```

Comment:
I don't think this should be a set. Let's make it a `Sequence` and then we can cast it to set a inside the function

Reply:
Ill make the update, but curious why not a `Set`?